_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -70,12 +86,51 @@ proc _cmake_get_build_dir {} {
     return [option worksrcpath]
 }
 
-default configure.dir {[_cmake_get_build_dir]}
-
-pre-configure {
-    file mkdir ${configure.dir}
+option_proc cmake.generator cmake::handle_generator
+proc cmake::handle_generator {option action args} {
+    global cmake.generator destroot destroot.target build.cmd build.post_args
+    global depends_build destroot.post_args build.jobs
+    if {${action} eq "set"} {
+        switch -nocase -glob ${args} {
+            "*Unix Makefiles*" {
+                ui_debug "Selecting the 'Unix Makefiles' generator"
+                cmake.generator     "Unix Makefiles"
+                depends_build-delete \
+                                    port:ninja
+                build.cmd           make
+                build.post_args     VERBOSE=ON
+                destroot.target     install/fast
+                destroot.destdir    DESTDIR=${destroot}
+                # unset the DESTDIR env. variable if it has been set before
+                destroot.env-delete DESTDIR=${destroot}
+            }
+            Ninja {
+                ui_debug "Selecting the Ninja generator"
+                cmake.generator     Ninja
+                depends_build-append \
+                                    port:ninja
+                build.cmd           ninja
+                # force Ninja to use the exact number of requested build jobs
+                build.post_args     -j${build.jobs}
+                destroot.target     install
+                # ninja needs the DESTDIR argument in the environment
+                destroot.destdir    ""
+                destroot.env-append DESTDIR=${destroot}
+            }
+            default {
+                if {[file tail ${cmake::currentportgroupdir}] eq "group"} {
```

Comment:
Usually you shouldn't have to add specific support for running this file from the registry. A certain state of a PortGroup will always be saved together with the Portfile used to install a port, so either it parsed fine at the install time, which means it will also parse fine when run from the registry, or it already threw this error on initial install.